### PR TITLE
service/dap: send continued event before step response

### PR DESCRIPTION
**Pull request by suzmue**
Send the continued event before the step response to make sure that there is no time where the client believes that only a single thread is running.

Fixes golang/vscode-go#1629
Updates golang/vscode-go#1617

### CODE REVIEW

File: service/dap/server.go [hunk 1]
```diff
@@ -1458,20 +1458,47 @@ func (s *Server) onAttachRequest(request *dap.AttachRequest) {
 // onNextRequest handles 'next' request.
 // This is a mandatory request to support.
 func (s *Server) onNextRequest(request *dap.NextRequest, asyncSetupDone chan struct{}) {
+	// All of the threads will be continued by this request, so we need to send
```
Comment (polinasok) [inline service/dap/server.go:1461]: how about a helper, so you don't repeat same code and comments 3 times?

Comment (suzmue) [reply service/dap/server.go]: Done.

**Comment by suzmue** [comment]
Fixes golang/vscode-go#1629

**Review by polinasok — APPROVED** [review]

**Comment by polinasok** [comment]
@derekparker Are we too late for 1.7.0 release? Vscode rolled out some UI fine tuning that exposed a timing bugs in our adapter that can be pretty annoying and confusing to the users.

**Review by aarzilli — APPROVED** [review]

**Pull request merged**